src/Provider/Provider.Web/Orchestrators/SearchHelperOrchestrator.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections;
+
+namespace Esfa.Recruit.Provider.Web.Orchestrators
+{
+    public class SearchHelperOrchestrator
+    {
+        const int maxCount = 50;
+        private readonly IProviderVacancyClient _providerVacancyClient;
+        public SearchHelperOrchestrator(IProviderVacancyClient providerVacancyClient)
+        {
+            _providerVacancyClient = providerVacancyClient;
+        }
+        public async Task<IEnumerable<string>> GetAutoCompleteListAsync(string term, long ukprn)
+        {
+            var result = new List<string>();
+
+            if (term == null || term.Trim().Length < 3) return result;
+
+            var vacancies = await GetVacanciesAsync(ukprn);
+
+            var data = vacancies
+                .Select(v => v.Title)
+                .Where(v => v.StartsWith(term, StringComparison.OrdinalIgnoreCase))
+                .Take(maxCount);
```

Comment:
Given that we are setting to take(maxcount), do we still need to check if data.count() < maxcount?

Reply:
if the query returns less than max then we need to fetch more suggestions.